### ISSUE TITLE
can't parse VALID ksh script with unterminated single quote

### ISSUE THREAD

**Issue by alanfranz** [koalaman/shellcheck#1475]
#### For bugs
* SC1072

```
$ shellcheck --version
ShellCheck - shell script analysis tool
version: 0.6.0
```
* The rule's wiki page does not already cover this
* I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/ksh

export ASD=`date +'%Y`
export OTHER=`date`

echo $ASD
echo $OTHER
```

#### Here's what shellcheck currently says:

```
$ shellcheck -s ksh -a basic.sh

In basic.sh line 3:
export ASD=`date +'%Y`
            ^-- SC1009: The mentioned syntax error was in this simple command.
                  ^-- SC1073: Couldn't parse this single quoted string. Fix to allow more checks.
                     ^-- SC1072: Expected end of single quoted string. Fix any mentioned problems and try again.

For more information:
  https://www.shellcheck.net/wiki/SC1072 -- Expected end of single quoted str...
  https://www.shellcheck.net/wiki/SC1073 -- Couldn't parse this single quoted...
  https://www.shellcheck.net/wiki/SC1009 -- The mentioned syntax error was in...
```

#### Here's what I wanted or expected to see:

As quirky as it may seem (I could not believe it myself, but then, I'm not a great ksh expert) **the above snippet works 100% in ksh**:

```
$ ksh ./basic.sh
2019
Thu Jan 24 17:48:54 CET 2019
```

So, I'd like shellcheck to allow whatever is allowed in ksh, go on parsing, and tell me what are further (and real) errors in that file.

Cheers!

**Comment by koalaman** [comment]
Haha, that's pretty amazing! I see that ksh and the original Bourne shell are the only shells that allow this, while Bash, Dash, Ash and Zsh give errors. This might explain why POSIX says "Either of the following cases produces undefined results: A single-quoted or double-quoted string that begins, but does not end, within the ``"`...`"`` sequence [...]"

In principle I agree, but given that this is not a good coding practice I doubt it's worth adding complexity to support it. Instead, it seems like a good first use case for ShellCheck's recoverable parsing mechanism.  With this change, you still get errors, but it'll continue checking the rest of the script. Thanks!
